crates/node/src/lib.rs

```diff
@@ -188,13 +201,18 @@ impl Node {
         Ok(())
     }
 
-    async fn attach_node(node_id: NodeId, cluster_controller_endpoint: Uri) -> Result<(), Error> {
-        info!("Attach to cluster at '{cluster_controller_endpoint}'");
-        let channel = Channel::builder(cluster_controller_endpoint.clone())
-            .connect_timeout(Duration::from_secs(5))
-            .connect_lazy();
+    async fn attach_node(
+        node_id: PlainNodeId,
+        cluster_controller_address: NetworkAddress,
+    ) -> Result<(), Error> {
```

Comment:
Will this return the NodeId in the future?

Reply:
Yes, this is the plan.